agent/command/git.go

```diff
@@ -308,6 +308,8 @@ func (c *gitFetchProject) buildCloneCommand(ctx context.Context, comm client.Com
 	gitCommands := []string{
 		"set -o xtrace",
 		"set -o errexit",
+		fmt.Sprintf("mkdir -p %s", c.Directory),
```

Comment:
This is actually a little interesting because in the original ticket, we stated that we may not have the necessary permissions to `rm -rf` the directory, so this `mkdir -p` can also fail for the same reasons as `rm -rf` due to permissions. If the directory structure doesn't exist yet, this will work, but if one of the parent directories _does_ exist and the agent doesn't have sufficient permissions to create a subdirectory within that directory, then `mkdir` will fail with a permissions error. For example:
```bash
mkdir -p parent # Make parent directory
chmod 111 parent # Make the directory read-only, causing later clone attempts to fail
// Later on, try cloning...

mkdir -p parent/child # Making the child directory will fail
chmod -R 755 parent/child # We won't get to chmod since the directory creation failed
```

Since we know this is a shell script, one thing we could do is move the `chmod` _before_ `set -o errexit`, so the script continues even if chmod fails due a nonexistent directory, and still fixes the permissions if it is able to. (And if we can't chmod for some other reason, we're doomed no matter what we try since we can't remove the directory at all).

If the above sounds reasonable, I'd still double-check in staging to make sure the script continues on error.

Reply:
Hm but if that's the case, if the parent directory exists but the child does not exist, we would need to do `chmod -R 755 parent` to make sure we can do mkdir -p parent/child inside that parent directory

Would something like:

chmod -R 755 dir/..
set -o errexit 
mkdir -p dir

Be good? Essentially, just moving the chmod above the set and pushing it back a directory?

Reply:
I guess my point is that `mkdir -p` doesn't really solve anything for this command when `chmod` is already a best-effort attempt to fix directory permissions. If I understand, the only reason `mkdir -p` is being used here is because the directory may not exist, causing `chmod` to fail. If you do the above ordering, then `mkdir -p` is no longer needed since you're about to remove the directory anyways.

As long as we have `chmod` continue on error (i.e. above `set -o errexit`), the mkdir is not needed. If it fails because the directory doesn't exist, it's fine because it'll continue on error. But if it fails because it's unable to change the permissions, then there's nothing we can do, and continuing on error is not a problem (`rm -rf` will fail later anyways, so the additional `mkdir -p` won't have any effect on git.get_project succeeding).

Reply:
Ahh that makes sense, ty!